language/functional_tests/tests/testsuite/commands/join_failure.mvir

```diff
@@ -1,10 +1,15 @@
 main() {
     let x: u64;
-    let y: &u64;
+    let y: u64;
+    let ref: &u64;
 
-    x = 7;
+    x = 6;
+    y = 7;
     if (true) {
-        y = &x;
+        ref = &x;
+    } else {
+        ref = &y;
+        _ = move(x);
```

Comment:
Does this test have anything to do with the references? Wouldn't deleting `ref`, `y`, and the then-branch also lead to a join failure?

Reply:
This is the check for `borrowed_value_unavailable`, i.e. "If there is a reference that is borrowing an empty local, join error"

In other words, the error is because in one branch we have `ref = &x`  in one branch and `move(x)` in another branch, which is currently invalid. 

But!
As I say this, I don't understand why its invalid, you never take both branches... I will confer offline with @shazqadeer